String.cxx

```diff
@@ -40,10 +40,8 @@
     return (m_u.compare(m_u.length() - s.m_u.length(), s.m_u.length(), s.m_u) == 0);
 }
 
-void String::getBytes(byte buffer[], unsigned length) const noexcept {
-    for (int i = 0; i < m_u.length; i++) {
-        buffer[i] = (byte)m_u[i];
-    }
+void String::getBytes(byte buffer[], unsigned length) const noexcept{
+    std::transform(m_u.begin(), m_u.end(), std::back_inserter(buffer), [](unsigned char c) -> byte { return c; });
```

Comment:
`std::back_inserter` on a `byte*` is straight up a compiler error
Also, `length` is unused, which indicates a bug